include/nbl/builtin/hlsl/glsl_compat/core.hlsl

```diff
@@ -135,6 +136,61 @@ float32_t2 unpackSnorm2x16(uint32_t p)
     return spirv::unpackSnorm2x16(p);
 }
 
+// No better way to get a vector type of same length as Integral but made up of `int32_t` s?
+
+template<typename T>
+vector<int32_t, uint32_t(sizeof(T) / sizeof(scalar_type_t<T>))> findLSB(T value)
+{
+    using return_t = vector<int32_t, uint32_t(sizeof(T) / sizeof(scalar_type_t<T>))>;
+    return bit_cast<return_t, T>(spirv::findILsb<T>(value));
+}
+
+namespace impl
+{
+    template<typename T, bool isSigned, bool isIntegral>
+    struct findMSB {};
+
+    template<typename T, bool isSigned>
+    struct findMSB<T, isSigned, false>
+    {
+        using return_t = vector<int32_t, uint32_t(sizeof(T) / sizeof(scalar_type_t<T>))>;
+
+        static return_t __call(T value)
+        {
+            static_assert(is_integral<T>::value, "T is not an integral type!");
+            return value;
+        }
+    };
+
+    template<typename T>
+    struct findMSB<T, true, true>
+    {
+        using return_t = vector<int32_t, uint32_t(sizeof(T) / sizeof(scalar_type_t<T>))>;
+
+        static return_t __call(T value)
+        {
+            return bit_cast<return_t, T>(spirv::findSMsb<T>(value));
+        }
+    };
+
+    template<typename T>
+    struct findMSB<T, false, true>
+    {
+        using return_t = vector<int32_t, uint32_t(sizeof(T) / sizeof(scalar_type_t<T>))>;
+
+        static return_t __call(T value)
+        {
+            return bit_cast<return_t, T>(spirv::findUMsb<T>(value));
+        }
+    };
+} // namespace impl
+
+template<typename T>
+vector<int32_t, uint32_t(sizeof(T) / sizeof(scalar_type_t<T>))> findMSB(T value)
+{
+    return impl::findMSB<T, is_signed_v<T>, is_integral_v<T> >::__call(value);
+}
+
```

Comment:
@Przemog1 I'll merge this PR before #801 , when resolving conflicts after merging `master` to 801, make sure to throw all this away.

Reply:
hehe turns out 801 got merged before this.